src/Uno.Extensions.Templates/content/unoapp-extensions/github/workflows/steps/build_ios/action.yml

```diff
@@ -0,0 +1,228 @@
+    required: false
+    default: '90'
+  bundle-identifier: 
+     description: ""
+     required: false
+     default: ''   
+  certificate: 
+     description: "Apple .p12 Base64 Certificate for signing final .ipa package."
+     required: true    
+     default: ''
+  certificate-password: 
+     description: "Apple .p12 certificate password."
+     required: true      
+     default: ''
+  codesign-key: 
+     description: "The name of the distribution certificate . Ex Apple Development: Contoso Contoso (752234A22L)"
+     required: true      
+     default: ''     
+  codesign-provision: 
+     description: "The provisioning profile name. This is the name you entered in the Apple Developer portal when creating your provisioning profile."
+     required: true      
+     default: ''          
+  provisioning-profile: 
+     description: "Base64 provisioning profile"
+     required: true
+     default: ''
+  keychain-password: 
+     description: "New keychain password. This can be a random string value"
+     required: false
+     default: 'KEYCHAIN_PASSWORD'      
+  publish-artifacts: 
+     description: "If true, compilation artifacts will be published and put under Job Summary section"
+     required: false
+     default: 'true' 
+  target-framework:
+     description: ''
+     required: false
+     default: 'net$ciTargetDotNetVersion$-ios'    
+runs:
+  using: "composite"
+  steps:
+    - name: Verify required parameters
+      shell: pwsh
+      run: |
+            $errors = ''
+            if ('' -eq '${{ inputs.project-to-build }}') { $errors = $errors + '[project-to-build] is not valid.' + "`r`n"  }
+            if ('' -eq '${{ inputs.certificate }}') { $errors = $errors +'[certificate] is not valid. Please make sure the name of the .p12 certificate is correct and it exists in secure files' + "`r`n"  }
+            if ('' -eq '${{ inputs.certificate-password }}') { $errors = $errors +'[certificate-password] is not valid. Please make sure you are using the correct .p12 certificate password' + "`r`n"  }
+            if ('' -eq '${{ inputs.codesign-key }}') { $errors = $errors +'[codesign-key] is not valid. Please make sure the name of the name of the distribution certificate is correct. E.g. Apple Development: Contoso Contoso (752234A22L)' + "`r`n"  }
+            if ('' -eq '${{ inputs.codesign-provision }}') { $errors = $errors +'[codesign-provision] is not valid. Please make sure the provisioning profile name is correct. This is the name you entered in the Apple Developer portal when creating your provisioning profile' + "`r`n"  }
+            if ('' -eq '${{ inputs.provisioning-profile }}') { $errors = $errors +'[provisioning-profile] is not valid. Please make sure the .mobileprovision file is correct and it exists in secure files' + "`r`n"  }
+            
+            if (-not ([string]::IsNullOrEmpty($errors))){
+                $errors
+                throw "Some required parameters are missing"
+            }
```

Comment:
scripts should be in a scripts directory where it is easier to get linting and intellisense and should not be inline like this when they are more than a couple of simple lines....

Reply:
@dansiegel  tried to include separate ps1 scripts but template did not let me to doing so because it represents a high security risk and that's why i included inline scripts instead which in theory represents the same risk but for some reason it let me do it that way ¯\_(ツ)_/¯